src/lib/encoder.ts

```diff
@@ -309,38 +311,38 @@ const decodeStringCompactBytesArray = (compactBytesArray: string): string[] => {
 
 const valueTypeEncodingMap = {
   bool: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value === true ? '0x01' : '0x00'),
```

Comment:
```suggestion
    encode: (value: boolean) => (value ? '0x01' : '0x00'),
```

this should work too